mantidproject/mantid
# Make fit work with tableworkspaces

**Description of work.**
This PR creates a function domain so that table workspaces will work with fit. 

**To test:**
The following script will create a table workspace with `ExpDecay` data and run the fit. The values in `fit_Parameters` should be: Height 10, LifeTime 1, Cost function very small (1e-27).  The data and calc values in the `fit_Workspace` should be approx the same and the diff should be around 0 for all x values.
```
import math

tableWS = CreateEmptyTableWorkspace()
tableWS.addColumn(type="double",name="X")
tableWS.addColumn(type="double",name="Y")
tableWS.addColumn(type="double",name="E")

for i in range(0,20):
      xValue = i * 0.1
      yValue = 10.0 * math.exp(-xValue)
      eValue = 0.01
      tableWS.addRow ( {'X': xValue, 'Y': yValue, 'E': eValue} )
    
myFunc = 'name=ExpDecay'
fit_output = Fit(InputWorkspace=tableWS, Output='fit', Function=myFunc, XColumn = 'X', YColumn = 'Y', ErrColumn = 'E')
paramTable = fit_output.OutputParameters
fitWorkspace = fit_output.OutputWorkspace
```

Fixes #26447 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
